bindata/v4.1.0/alerts/audit-errors.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: audit-errors
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+  - name: apiserver-audit
+    rules:
+    - alert: AuditLogError
+      annotations:
+        summary: |-
+          An API Server instance was unable to write audit logs. This could be
+          triggered by the node running out of space, or a malicious actor
+          tampering with the audit logs.
+        description: An API Server had an error writing to an audit log.
+      expr: |
+        sum by (apiserver,instance)(rate(apiserver_audit_error_total{apiserver=~".+-apiserver"}[5m])) / sum by (apiserver,instance) (rate(apiserver_audit_event_total{apiserver=~".+-apiserver"}[5m])) > bool 0
```

Comment:
I don't think that you want the `bool` modifier because the alert will fire forever once `apiserver_audit_error_total` is initialized/exposed (e.g. the alert will be active even if `sum by (apiserver,instance)(rate(apiserver_audit_error_total{apiserver=~".+-apiserver"}[5m]))` is 0 the full expression will return 0).

```suggestion
        sum by (apiserver,instance)(rate(apiserver_audit_error_total{apiserver=~".+-apiserver"}[5m])) / sum by (apiserver,instance) (rate(apiserver_audit_event_total{apiserver=~".+-apiserver"}[5m])) > 0
```

From https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1166/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/1410634018554122240, we can see that `apiserver_audit_error_total` isn't present but I bet that it appears only after the first error is encountered.

![image](https://user-images.githubusercontent.com/2587585/124258036-41453d00-db2d-11eb-832d-03b003489f03.png)

Reply:
That's right, it's not present until there's an issue... and it gets even trickier because say you `rm` or `chmod` the `audit.log` file: The Kubernetes API Server won't detect that until it attempts to rotate the file. But that's an issue in Kubernetes that has to be addressed in another way.

Reply:
@simonpasquier removed the `bool` as suggested.

Reply:
thanks!